src/docs/release/breaking-changes/eliminating-nullok-parameters.md

```diff
@@ -0,0 +1,266 @@
+---
+
+## Summary
+
+This migration guide describes conversion of code that uses the `nullOk`
+parameter on multiple `of` static accessors and related accessors to use
+alternate APIs with nullable return values.
+
+## Context
+
+Flutter has a common pattern of allowing lookup of some types of widgets
+([`InheritedWidget`]s) using static member functions that are typically called
+`of`, and take a `BuildContext`.
+
+Before non-nullability was the default, it was useful to have a toggle on these
+APIs that swapped between throwing an exception if the widget was not present in
+the widget tree and returning null if it was not found. It was useful, and
+wasn't confusing, since every variable was nullable.
+
+When non-nullability was made the default, it was then desirable to have the
+most commonly used APIs return a non-nullable value. This is because saying
+`MediaQuery.of(context, nullOk: false)` and then still requiring an `!` operator
+or `?` and a fallback value after that call felt awkward.
+
+The `nullOk` parameter was a cheap form of providing a null safety toggle, which
+in the face of true language support for non-nullability, was then supplying
+redundant, and perhaps contradictory signals to the developer.
+
+To solve this, the `of` accessors (and some related accessors that also used
+`nullOk`) were split into two calls: one that returned a non-nullable value and
+threw an exception when the sought-after widget was not present, and one that
+returned a nullable value that didn't throw an exception, and returned null if
+the widget was not present.
+
+The design document for this change is here:
+[https://flutter.dev/go/eliminating-nullok-parameters](
+https://flutter.dev/go/eliminating-nullok-parameters)
+
+## Description of change
+
+The actual change modified these APIs to not have a `nullOk` parameter, and to
+return a non-nullable value:
+
+1. [`MediaQuery.of`]
+1. [`Navigator.of`]
+1. [`ScaffoldMessenger.of`]
+1. [`Scaffold.of`]
+1. [`Router.of`]
+1. [`Localizations.localeOf`]
+1. [`FocusTraversalOrder.of`]
+1. [`FocusTraversalGroup.of`]
+1. [`Focus.of`]
+1. [`Shortcuts.of`]
+1. [`Actions.handler`]
+1. [`Actions.find`]
+1. [`AnimatedList.of`]
+1. [`SliverAnimatedList.of`]
+1. [`CupertinoDynamicColor.resolve`]
+1. [`CupertinoDynamicColor.resolveFrom`]
+1. [`CupertinoUserInterfaceLevel.of`]
+1. [`CupertinoTheme.brightnessOf`]
+1. [`CupertinoThemeData.resolveFrom`]
+1. [`NoDefaultCupertinoThemeData.resolveFrom`]
+1. [`CupertinoTextThemeData.resolveFrom`]
+1. [`MaterialBasedCupertinoThemeData.resolveFrom`]
+
+And introduced these new APIs alongside those, to return a nullable
+value:
+
+1. [`MediaQuery.maybeOf`]
+1. [`Navigator.maybeOf`]
+1. [`ScaffoldMessenger.maybeOf`]
+1. [`Scaffold.maybeOf`]
+1. [`Router.maybeOf`]
+1. [`Localizations.maybeLocaleOf`]
+1. [`FocusTraversalOrder.maybeOf`]
+1. [`FocusTraversalGroup.maybeOf`]
+1. [`Focus.maybeOf`]
+1. [`Shortcuts.maybeOf`]
+1. [`Actions.maybeHandler`]
```

Comment:
Running an audit to double check from last night, I can't find Actions.maybeHandler either

Reply:
Looks like this is intentional per the change description: https://github.com/flutter/flutter/pull/68921
```suggestion
```

Reply:
Yes, removed.